### PR TITLE
BAU: document identity flow

### PR DESCRIPTION
**Pull request by ethanmills**
## Checklist

- [ ] Impact on orch and auth mutual dependencies has been checked.

- [ ] A UCD review has been performed.

### CODE REVIEW

File: docs/diagrams/orchestration/identity/identity.mmd [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    auth ->> auth : Authenticates the user
+    auth -->> orch_api : Return the user with an auth code query parameter
+    Note over orch_api,auth: Solid lines indicate machine-machine API calls
+    orch_api ->> auth : Token request containing auth code to /token
+    auth ->> orch_api : Token response containing access token
+    orch_api ->> auth : Userinfo request containing access token to /userinfo
+    auth ->> orch_api : Userinfo response
+    orch_api -->> ipv : OAuth 2.0 Authorization request to /authorize
+    ipv -->> orch_api : Return the user with an auth code query parameter
+    orch_api ->> ipv : Exchange the auth code for an access token with a signed client assertion
```
Comment (billrdunn) [inline docs/diagrams/orchestration/identity/identity.mmd:22]:
```suggestion
    orch_api ->> ipv : Token request (with signed client assertion) containing auth code to /token
```

This and below suggestions just to be consistent with the wording of the auth flow above

File: docs/diagrams/orchestration/identity/identity.mmd [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    ipv ->> orch_api : Issue an access token
```
Comment (billrdunn) [inline docs/diagrams/orchestration/identity/identity.mmd:23]:
```suggestion
    ipv ->> orch_api : Token response containing access token
```

File: docs/diagrams/orchestration/identity/identity.mmd [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    orch_api ->> ipv : Request for user identity authorised with the access token
```
Comment (billrdunn) [inline docs/diagrams/orchestration/identity/identity.mmd:24]:
```suggestion
    orch_api ->> ipv : Useridentity request containing access token to /useridentity
```

File: docs/diagrams/orchestration/identity/identity.mmd [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    ipv ->> orch_api : User identity response
```
Comment (billrdunn) [inline docs/diagrams/orchestration/identity/identity.mmd:25]:
```suggestion
    ipv ->> orch_api : Useridentity response
```

File: docs/diagrams/orchestration/identity/identity.mmd [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    
+
+    orch_api ->> orch_api : Generates and store Authorization Code with 5 min TTL
+    orch_api ->> auth : Generate a redirect URL containing the auth code
+
+    auth -->> rp : Redirect the user to the redirect URL
+    rp ->> orch_api : Token request to /token
+    orch_api ->> orch_api : Validates token request, then generates and stores access token with 3 min TTL
+    orch_api ->> rp : Token response containing access and id tokens
+    rp ->> orch_api : UserInfo request to /userinfo
```
Comment (billrdunn) [inline docs/diagrams/orchestration/identity/identity.mmd:47]:
```suggestion
    rp ->> orch_api : Userinfo request containing access token to /userinfo
```

File: docs/diagrams/orchestration/identity/identity.mmd [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    orch_api ->> orch_api : Validates Access Token
+    orch_api ->> rp : UserInfo response containing user information, including signed Core Identity JWT and any additional permitted identity claims
```
Comment (billrdunn) [inline docs/diagrams/orchestration/identity/identity.mmd:49]:
```suggestion
    orch_api ->> rp : Userinfo response containing user information, including signed Core Identity JWT and any additional permitted identity claims 
```

File: docs/diagrams/orchestration/identity/identity.mmd [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+    orch_api ->> orch_api : Generates and store Authorization Code with 5 min TTL
+    orch_api ->> auth : Generate a redirect URL containing the auth code
+
+    auth -->> rp : Redirect the user to the redirect URL
+    rp ->> orch_api : Token request to /token
+    orch_api ->> orch_api : Validates token request, then generates and stores access token with 3 min TTL
+    orch_api ->> rp : Token response containing access and id tokens
+    rp ->> orch_api : UserInfo request to /userinfo
+    orch_api ->> orch_api : Validates Access Token
```
Comment (billrdunn) [inline docs/diagrams/orchestration/identity/identity.mmd:48]:
```suggestion
    orch_api ->> orch_api : Validates access token
```

File: docs/diagrams/orchestration/identity/identity.mmd [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    auth ->> orch_api : Poll for updates
+    orch_api ->> auth : COMPLETED response
+    auth ->> orch_api : Request an RP auth code
+    
+
+    orch_api ->> orch_api : Generates and store Authorization Code with 5 min TTL
+    orch_api ->> auth : Generate a redirect URL containing the auth code
+
+    auth -->> rp : Redirect the user to the redirect URL
+    rp ->> orch_api : Token request to /token
```
Comment (billrdunn) [inline docs/diagrams/orchestration/identity/identity.mmd:44]:
```suggestion
    rp ->> orch_api : Token request containing auth code to /token
```

**Review by billrdunn — COMMENTED** [review]
Thanks for making this, very useful. Sorry about the nitpicks :)

File: docs/diagrams/orchestration/identity/identity.mmd (hunk 4 above)
Comment (ethanmills) [reply docs/diagrams/orchestration/identity/identity.mmd:25]: I think I prefer this with the space. It's not a concept in the same way as `userinfo` is

File: docs/diagrams/orchestration/identity/identity.mmd (hunk 5 above)
Comment (ethanmills) [reply docs/diagrams/orchestration/identity/identity.mmd:47]: This is the way it's formatted in the spec https://connect2id.com/products/server/docs/api/userinfo

Comment (billrdunn) [reply docs/diagrams/orchestration/identity/identity.mmd]: Sure, will need to update the other references to be `UserInfo`

Comment (ethanmills) [reply docs/diagrams/orchestration/identity/identity.mmd]: I've updated this

**Review by billrdunn — APPROVED** [review]

**Pull request merged**